pkg/storage/unified/resource/server.go

```diff
@@ -1037,6 +1117,14 @@ func (s *server) GetStats(ctx context.Context, req *ResourceStatsRequest) (*Reso
 	if err := s.Init(ctx); err != nil {
 		return nil, err
 	}
+
+	if s.shardingEnabled {
+		client := s.getClientToDistributeRequest(req.Namespace)
+		if client != nil {
+			return client.Client.GetStats(userutils.InjectOrgID(ctx, "1"), req)
```

Comment:
nit: Instead of injecting dummy OrgID, we could remove `middleware.ClientUserHeaderInterceptor` from gRPC clients, and auth check for GetStats method.